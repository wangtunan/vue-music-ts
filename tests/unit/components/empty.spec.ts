import { shallowMount, Wrapper } from '@vue/test-utils'
import Empty from '@/components/empty/index.vue'

describe('empty.vue', () => {
  let wrapper: Wrapper<Vue>
  beforeEach(() => {
    wrapper = shallowMount(Empty)
  })
  it('match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('default props.title', () => {
    expect(wrapper.props('title')).toBe('抱歉，暂无数据')
  })
  it('pass props.title', () => {
    const title = '暂时没有相关数据'
    wrapper = shallowMount(Empty, {
      propsData: {
        title
      }
    })
    expect(wrapper.find('.title').text()).toBe(title)
  })
})