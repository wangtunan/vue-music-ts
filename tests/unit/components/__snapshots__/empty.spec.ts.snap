// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty.vue match snapshot 1`] = `
<div class="m-empty"><img src="" alt="" class="image">
  <p class="title">抱歉，暂无数据</p>
</div>
`;
